packages/cli/src/util/dev/server.ts

```diff
@@ -2082,6 +2085,27 @@ export default class DevServer {
     }
   };
 
+  private logRequestResponse(
+    req: http.IncomingMessage,
+    res: http.ServerResponse
+  ) {
+    const method = (req.method || 'GET').padEnd(4);
```

Comment:
Nit pick: should `4` be `7` instead since there are [methods](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods) like `OPTIONS` that are 7 chars long?

Reply:
I picked 4 because get/post seem most likely. If a longer one comes along it will no longer be aligned, but I think that's fine over every GET having 4 extra spaces.

I could go either way, though.

Reply:
Seems like it should either be aligned or not. So either drop pad or make it work for all methods. There are so few methods its going to look like a bug if its aligned on some but not others

Reply:
Not aligned at all:

<img width="424" alt="CleanShot 2023-02-16 at 16 11 13@2x" src="https://user-images.githubusercontent.com/41545/219499833-8e34590c-d04a-489d-9eaa-a14b67eed8e1.png">

Aligned to 7 spaces:
<img width="495" alt="CleanShot 2023-02-16 at 16 16 28@2x" src="https://user-images.githubusercontent.com/41545/219499894-b52928b4-8945-4ac5-8b29-556bbcec737d.png">

---

I like 7. What do you think? [`795c863` (#9411)](https://github.com/vercel/vercel/pull/9411/commits/795c86390522856732a625e663fd5bcd34470568)

Reply:
Agreed lets go with 7.

Also, are we still planning to log for success? I though we decided to only log on error?

Reply:
We don't know a request will be successful until it responds. If we wait for the response to show any log message, it will feel slow. I think we want to see the request as soon as it was triggered.

We were logging all requests and all responses before, but I changed it earlier to:

- request
    - lambda or edge: log
    - else: hide
- response
    - lambda or edge:
        - <400: hide
        - 404: hide
        - else: log
    - else: hide
    
Relevant commits:

- [`ac3acf1` (#9411)](https://github.com/vercel/vercel/pull/9411/commits/ac3acf1a5ea6c2a5451de89c82aee03deffde01d)
- [`289e6cb` (#9411)](https://github.com/vercel/vercel/pull/9411/commits/289e6cb3c84a8d5de83b0e3eade8a3e692d77d27)

Reply:
Is it going to be too noise and hide the user's `console.log()`? Is there a way to turn it off if the user wants to only see their own logs?

Reply:
Potentially.

Option 1 - Dim: We could dim all the log lines except the exceptional ones. That would make the default text of a user's `console.log()` stand out.

Option 2 - Flag: We could add an option to disable this behavior. It may make sense to go fully into the realm of setting log levels.

Option 3 - Error Only: We could not add these request/response logs in this way at all.

Reply:
I'm +1 for Option 2 - implementing a log level system. It would default to `warn` which would also display errors. A new global `--log-level <level>` option makes sense, but I would really like to be able to control the log level using an environment variable. Users could set the env var globally or in the `.env` file.

Reply:
What would be the possible values of `--log-level`?

Reply:
Traditionally the value would be a string containing one of:
* `'error'` - Errors only
* `'warn'` - Errors and warnings
* `'info'` - General useful info (URLs, ports, something has started)
* `'debug'` - Dumping variables, general debug logging

I've also see `'trace'` that would be ultra detailed output such as `stdout` from spawned processes, etc, but I don't think we need this.

Reply:
Chrome's Dev Tools console has `'error'`, `'warn'`, `'info'`, and `'verbose'`. I kinda like `'verbose'` which essentially means no filtering, show everything.

Reply:
Since we already have —debug, perhaps that existing feature can be merged with this new feature

Reply:
Yeah, I would fold in `--debug` as a shorthand for `--loglevel=debug`.

Should we block this PR on implementing better log level management? I'd be fine with that.

Reply:
We discussed this on a call:

- put these logs behind `--info` so that they are not shown by default
- this is compatible with a future change where logs levels are managed more explicitly
- restore the function path in the error messages in case other log lines (perhaps from frontend server) end up between request/response log lines

Reply:
If I use `--debug`, do I also see `--info` logs plus more?

Reply:
Right now, that would mean you'd see two different representations of a request log line, but that's probably OK until we revise log levels later. When we add log levels, we could audit what is shown in the `debug` level and possibly remove the `debug` request log line.

For this PR, I think it'd be fine to show both in `--debug`.

Reply:
We can leave them separate for minimal disruption, then when we implement full log level support we do make them cascade. I'm good with either, though.